aggregator/src/aggregator.rs

```diff
@@ -264,33 +264,51 @@ impl<C: Clock> Aggregator<C> {
     ) -> Result<HpkeConfigList, Error> {
-        match task_id_base64 {
-            Some(task_id_base64) => {
-                let task_id_bytes = URL_SAFE_NO_PAD
-                    .decode(task_id_base64)
-                    .map_err(|_| Error::UnrecognizedMessage(None, "task_id"))?;
-                let task_id = TaskId::get_decoded(&task_id_bytes)
-                    .map_err(|_| Error::UnrecognizedMessage(None, "task_id"))?;
-                let task_aggregator = self
-                    .task_aggregator_for(&task_id)
-                    .await?
-                    .ok_or(Error::UnrecognizedTask(task_id))?;
-                Ok(task_aggregator.handle_hpke_config())
+        // If we're running in taskprov mode, unconditionally provide the global keys and ignore
+        // the task_id parameter.
+        if self.cfg.taskprov_config.enabled {
+            let configs = self.global_hpke_keypairs.configs();
+            if configs.is_empty() {
+                Err(Error::Internal(
+                    "this server is missing its global HPKE config".into(),
+                ))
+            } else {
+                Ok(HpkeConfigList::new(configs.to_vec()))
             }
-            None => {
-                let configs = self.global_hpke_keypairs.configs();
-                if configs.is_empty() {
-                    if self.cfg.taskprov_config.enabled {
-                        // A global HPKE configuration is only _required_ when taskprov
-                        // is enabled.
-                        Err(Error::Internal(
-                            "this server is missing its global HPKE config".into(),
-                        ))
-                    } else {
+        } else {
+            // Otherwise, try to get the task-specific key.
+            match task_id_base64 {
+                Some(task_id_base64) => {
+                    let task_id_bytes = URL_SAFE_NO_PAD
+                        .decode(task_id_base64)
+                        .map_err(|_| Error::UnrecognizedMessage(None, "task_id"))?;
+                    let task_id = TaskId::get_decoded(&task_id_bytes)
+                        .map_err(|_| Error::UnrecognizedMessage(None, "task_id"))?;
+                    let task_aggregator = self
+                        .task_aggregator_for(&task_id)
+                        .await?
+                        .ok_or(Error::UnrecognizedTask(task_id))?;
+
+                    match task_aggregator.handle_hpke_config() {
+                        Some(hpke_config_list) => Ok(hpke_config_list),
+                        // Assuming something hasn't gone horribly wrong with the database, this
+                        // should only happen in the case where the system has been moved from taskprov
+                        // mode to non-taskprov mode. Thus there's still taskprov tasks in the database.
+                        // This isn't a supported use case, so the operator needs to delete these tasks
```

Comment:
I've updated https://github.com/divviup/janus/pull/1736 to back up this assertion.